esx_service/utils/auth_api.py

```diff
@@ -491,21 +491,43 @@ def check_datastore(datastore_name):
     if not datastore_exist(datastore_name):
         error_info = error_code.generate_error_info(ErrorCode.DS_NOT_EXIST, datastore_name)
         return error_info
-    
+
     return None
 
 def privilege_exist(privileges, datastore_name):
-    """ Check wheter a entry with given datastore_name exists in privileges """
+    """ Check whether a entry with given datastore_name exists in privileges """
     for p in privileges:
         ds_name = vmdk_utils.get_datastore_name(p.datastore_url)
         if ds_name == datastore_name:
             return True
-    
+
     return False
-    
+
+def check_privilege_parameters(privilege):
+    """ Check whether the privilege parameters are invalid """
+
+    volume_maxsize = privilege[auth_data_const.COL_MAX_VOLUME_SIZE]
+    volume_totalsize = privilege[auth_data_const.COL_USAGE_QUOTA]
+
+    # If both volume max size and volume total size are set,
+    # volume max size should not exceed volume total size
+    if (volume_maxsize and volume_totalsize and (volume_maxsize > volume_totalsize)):
+        error_msg = "Volume max size {0}MB exceeds the total size {1}MB".format(
+            volume_maxsize, volume_totalsize)
+        error_info = error_code.generate_error_info(ErrorCode.INVALID_ARGUMENT, error_msg)
+        return error_info
+
+    # If volume total size is set, but volume max size is NOT set,
+    # volume max size should be set to the same as volume total size
+    if (volume_totalsize and not volume_maxsize):
+        print('Note: volume max size will be set to the same value as volume total size: {0}MB'.format(volume_totalsize))
```

Comment:
We should not use print here. Is it a error case here? It it is, return the error, otherwise, we should log this.
I don't understand why you assign volume_maxsize to be volume_totalsize here. There two parameters are independent.